redis/redis.go

```diff
@@ -30,7 +33,10 @@ func (c Client) Set(k string, v interface{}) error {
 		return err
 	}
 
-	err = c.c.Set(k, string(data), 0).Err()
+	tctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
```

Comment:
I would prefer one of the following:
- Not break old behavior, use context without timeout
- Extend the config with a configurable timeout. Some gokv packages already do that, like `etcd` one [here](https://github.com/philippgille/gokv/blob/93d491e1a0c9ba21a1a88e7e5888da159671a122/etcd/etcd.go#L101)

And in the future I think we should change `gokv` signatures to one with a context as first parameter, following best practices and giving users full control of per-call timeouts.

Reply:
thanks, sure I can add timeout config!

I was following https://github.com/philippgille/gokv/blob/99127ded57972e470ac9d4a88008c0c9f4d8b8a6/dynamodb/dynamodb.go#L254 and https://github.com/philippgille/gokv/blob/99127ded57972e470ac9d4a88008c0c9f4d8b8a6/datastore/datastore.go#L45 that have hard-coded timeout, but I can follow https://github.com/philippgille/gokv/blob/93d491e1a0c9ba21a1a88e7e5888da159671a122/etcd/etcd.go#L101 and also update the other two to add timeout configuration

Yes, context as first parameter is much needed at this point!

Reply:
Yes indeed, not all store implementations have a configurable timeout yet. Thanks for the update!